dogfood.md

```diff
@@ -0,0 +1,167 @@
+
+A meme consists of a picture, a comment displayed aligned to the top of the picture, and a comment displayed aligned with the bottom of the picture.  
+
+### Users can delete their activity
+A user should have the ability to delete all of their comments and meme history from the app.  When a user deletes a meme it will also delete all comments associated with that meme
+
+### Where possible offline usage should be supported
+
+Users should be able to create memes, view cached memes, author comments, and like posts while offline.  Once the device returns online the qued activity should be submitted to the server even if the app is backgrounded.
+
+## Clients
+
+The memeolist dogfood exercise will support iOS and Android native as well as cross platform development with our Cordova and Xamarin SDKs.  Additionally, the use of GraphQL should let us generate some of the client code which means we won't be starting from scratch for each platform.
+
+### Android
+
+David Martin has done some work with graphQL and Android related to the current research on a new sync service.  The Apollo project has good tooling for supporting GraphQL APIs on Android, and the AeroGear Android SDK supports all of the features we are planning on targetting.  The existing memeolist code from the FeedHenry demo can also serve as a starting point for development and testing, though it will need to undergo a lot of revisions.
+
+### Xamarin SDK
+
+The Xamarin SDK is lacking support for UPS which will have to be implemented for both iOS and Android.  Additionally we lack experience with Xamarin and GraphQL, but a Google search shows that there are a few projects which can be researched for support.  This application will also offer people interested in upskilling on Xamarin a chance to work on a project from greenfield.
+
+### Cordova SDK
+GraphQL seems to have a lot of javascript support which will be useful for our Javascript/Cordova client application.  The Cordova SDK has all of the features we will use, and we have a lot of JavaScript expertise to draw on.
+
+### iOS
+
+The iOS SDK has all of the features we need, and iOS has an Apollo GraphQL library as well.  Like Cordova and Android there is a lot of experience we can draw on for developing the app.
+
+
+## Data Model and API layer
+
+The following is a graphql schema for the proposed data model.  The GraphQL schema also includes the public APIs that will be implemented.
+
+### Sample Schema
+
+```graphql
+type Profile {
+  id: ID! @unique
+  email: String!
+  display_name: String!
+  biography: String!
+  pictureUrl: String!
+}
+
+type MemeList {
+  id: ID! @unique
+  ownerId: ID!
+  createdOn: String!
+  updatedOn: String!
+  memes: [Meme!]!
+}
+
+type Meme {
+  id: ID! @unique
+  ownerId: ID!
+  createdOn: String!
+  updatedOn: String!
+  photoUrl: String!
```

Comment:
@darahayes @aliok the first pass at getting the memolist app to use graphql/data sync will just have the photoUrl of the meme, rather than the binary content.
The image upload will be handled on a separate channel i.e. a separate service specific to memeolist with some http multipart upload endpoint.

However, binary upload via graphql is an ask on the board, which we should look at after an initial end to end with the memolist app and data sync.https://trello.com/c/SzOJqLqA/8-as-a-developer-i-want-the-ability-to-synchronise-binary-data-eg-photos-videos-audio-files-via-the-sync-framework

Reply:
@david-martin maybe we can do that transforming the file into a byte[]/String using Base64 to be stored as a "string"? It can be handled by the SDK

Reply:
See comment here for response https://github.com/aerogear/proposals/pull/40#discussion_r198660830